src/test/java/codeu/controller/ChatServletTest.java

```diff
@@ -211,4 +214,52 @@ public void testDoPost_CleansHtmlContent() throws IOException, ServletException
 
     Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
   }
+
+  @Test
+  public void testDoPost_makePublicPrivate() throws IOException, ServletException {
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/chat/test_conversation");
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+    Mockito.when(mockUserStore.getUser("test_username")).thenReturn(
+        new User(
+          UUID.randomUUID(),
+          "test_username",
+          "testHash",
+          Instant.now()));
+    Mockito.when(mockConversationStore.getConversationWithTitle("test_conversation"))
+        .thenReturn(mockConversation);
+
+    Mockito.when(mockRequest.getParameter("type"))
+        .thenReturn("private");
+    Mockito.when(mockConversation.getType()).thenReturn("public");
+    
+    chatServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockConversation).makePrivate();
+    Mockito.verify(mockConversationStore).updateConversation(mockConversation);
+    Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
+  }
+
+  @Test
+  public void testDoPost_makePrivatePrivate() throws IOException, ServletException {
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/chat/test_conversation");
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+    Mockito.when(mockUserStore.getUser("test_username")).thenReturn(
+        new User(
+          UUID.randomUUID(),
+          "test_username",
+          "testHash",
+          Instant.now()));
+    Mockito.when(mockConversationStore.getConversationWithTitle("test_conversation"))
+        .thenReturn(mockConversation);
+
+    Mockito.when(mockRequest.getParameter("type"))
+        .thenReturn("private");
+    Mockito.when(mockConversation.getType()).thenReturn("private");
+    
+    chatServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockConversation, Mockito.never()).makePrivate();
```

Comment:
Perhaps we should just test that the conversation object isn't public at the end of the test, rather than verifying the mocks have/haven't been called?

Reply:
My goal was to test whether or not private conversation would be updated redundantly when a users clicks the private option on an already private conversation. is that not a relevant case to have a test for? and sorry about the test name, it could've been much clearer.

Reply:
Although now that I'm implementing private -> publc, I will change up the tests a bit